src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -92,7 +93,11 @@ public function getTypeExtensions()
      */
     public function createBuilder(FormFactoryInterface $factory, $name, array $options = [])
     {
-        $options = $this->getOptionsResolver()->resolve($options);
+        try {
+            $options = $this->getOptionsResolver()->resolve($options);
+        } catch (ExceptionInterface $e) {
+            throw new $e(sprintf('An error has occurred resolving the options of the form "%s": %s', \get_class($this->getInnerType()), $e->getMessage()), $e->getCode(), $e);
```

Comment:
that's legal? wow :)
are we sure all constructors can be called with the message first?

Reply:
Yes, at least all those who implement `ExceptionInterface` so far.